flink-runtime/src/main/java/org/apache/flink/runtime/scheduler/SchedulerBase.java

```diff
@@ -286,7 +286,7 @@ private void tryRestoreExecutionGraphFromSavepoint(ExecutionGraph executionGraph
 		}
 	}
 
-	protected void resetForNewExecutions(final Collection<ExecutionVertexID> vertices) {
+	protected void resetForNewExecutions(final List<ExecutionVertexID> vertices) {
```

Comment:
Is this needed?

Reply:
I think it's better to regard it as ordered. See my other comment.

Reply:
we do not sort `tasksToRestart` now and this change is dropped since it is not very necessary right now.